### PR TITLE
Favorite functionality 

### PR DESCRIPTION
**Pull request by jacobogart**
- Add item to favorites
- Remove items from favorites
- Heart on card is red if that object is favorited

### CODE REVIEW

**Comment by jacobogart** [comment]
Issues when moving directly from one card to another:

- heart does not render in correct color
- loop hole allowing an item to be favorited more than once

File: src/components/App.js [hunk 1]
```diff
@@ -80,12 +89,13 @@ class App extends React.Component {
...
       Shows: searchShows("", this.state.bars),
-      Queens: searchQueens("", this.state.queens)
+      Queens: searchQueens("", this.state.queens),
+      Favorites: this.state.favoritesList || []
     };
     this.setState({
       searchResults: searches[type]
-    });
-  };
+    }, this.displayAllSearchResults());
```
Comment (SiimonStark) [inline src/components/App.js:97]: I assume the ` }, this.displ... ` is triggering our showAllResults function directly after setting the searchQuery to state?
@jacobogart

File: src/components/Card.js [hunk 2]
```diff
@@ -23,13 +23,70 @@ export default class Card extends Component {
...
     };
   }
+
+  componentDidMount() {
+    this.checkFavorite();
+  }
+
+  checkFavorite = () => {
+    if (localStorage.getItem("favoritesList")){
+      let localFavoritesList = JSON.parse(localStorage.getItem("favoritesList")).map(object => object.name);
```
Comment (SiimonStark) [inline src/components/Card.js:36]: You knocked this one out SOO fast!
I helped a lil bit on this one!! 😁

File: src/components/Card.js [hunk 3]
```diff
@@ -23,13 +23,70 @@ export default class Card extends Component {
...
+      this.removeFromFavorites();
+    } else {
+      this.addToFavorites();
+    }
+    this.setState({
+      favorite: !this.state.favorite
+    });
+  }
+
+  removeFromFavorites = () => {
```
Comment (SiimonStark) [inline src/components/Card.js:56]: Wow, great logic on here! Not sure how you figured this out so quickly! 🧠
I like how you're using the name as the index in order to remove the Object from LocalStorage

Comment (SiimonStark) [reply src/components/Card.js:56]: @KevinKra please be sure to thoroughly review the Local Storage bits of this code!
It's really sound logic that could easily be implemented in other use cases 👍🏻

File: src/components/Card.js [hunk 4]
```diff
@@ -23,13 +23,70 @@ export default class Card extends Component {
...
     const cardData = this.props.cardData;
     const { id } = cardData;
     let mainInfo, cardType;
+    let favorite = false;
+    
+    if (localStorage.getItem("favoritesList")) {
+      let localFavoritesList = JSON.parse(localStorage.getItem("favoritesList")).map(object => object.name);
+      if (localFavoritesList.includes(this.props.cardData.name)) {
+          favorite = true
+      }
```
Comment (SiimonStark) [inline src/components/Card.js:88]: Would rounding off this conditional with an ` else favorite = false ` help us fix the bug where the heart stays red switching from card to card?
@jacobogart

File: src/tests/SearchResults.test.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
```
Comment (SiimonStark) [inline src/tests/SearchResults.test.js:1]: Tomorrow we'll need to run npm --coverage to check how many more tests we need. I have feeling we may be missing a few

**Review by SiimonStark — APPROVED** [review]
Outstanding work on nailing the Local Storage! The bugs are reasonable, and hopefully we can flush them out tomorrow!

File: src/components/App.js (hunk 1 above)
Comment (jacobogart) [reply src/components/App.js:97]: Yes, I just piggybacked off the search Results functionality for all the hamburger menu functions

File: src/components/Card.js (hunk 4 above)
Comment (jacobogart) [reply src/components/Card.js:88]: Right now there is both a favorite state that determines if the element needs to be added or removed and a local favorites variable that determines if the heart should be red or not. This logic just evaluates if the heart should be red, which is set at false at the top of the render. But I guess an else wouldn't hurt.

**Pull request merged**